### PR TITLE
AzureCI changes. RVSDG test config should still test its assigned test slice

### PR DESCRIPTION
**Pull request by sklam**
Without the patch, a test slice is skipped entirely.

### CODE REVIEW

**Review by stuartarchibald — APPROVED** [review]
Thanks for finding and fixing this @sklam, this should restore full shard coverage.

File: buildscripts/incremental/test.sh [hunk 1]
```diff
@@ -119,32 +119,32 @@ if [[ $(uname) == "Darwin" ]]; then
...
+    coverage erase
+    $SEGVCATCH coverage run runtests.py -b -j "$TEST_START_INDEX:$TEST_COUNT" --exclude-tags='long_running' -m $TEST_NPROCS -- numba.tests
+elif [ "$RUN_TYPEGUARD" == "yes" ]; then
+    echo "INFO: Running with typeguard"
+    NUMBA_USE_TYPEGUARD=1 NUMBA_ENABLE_CUDASIM=1 PYTHONWARNINGS="ignore:::typeguard" $SEGVCATCH python runtests.py -b -j "$TEST_START_INDEX:$TEST_COUNT" --exclude-tags='long_running' -m $TEST_NPROCS -- numba.tests
+else
+    NUMBA_ENABLE_CUDASIM=1 $SEGVCATCH python -m numba.runtests -b -j "$TEST_START_INDEX:$TEST_COUNT" --exclude-tags='long_running' -m $TEST_NPROCS -- numba.tests
+fi
+
+# Now run with RVSDG enabled for a subset of tests
```
Comment (stuartarchibald) [inline buildscripts/incremental/test.sh:147]: I think the if needs a corresponding `fi` ?

Comment (sklam) [reply buildscripts/incremental/test.sh:147]: we saw it at the same time. fixed in https://github.com/numba/numba/pull/9360/commits/2461cfb9df2919044bda6bedf33bd1c699fd2a87

**Comment by sklam** [comment]
Expected failure on https://dev.azure.com/numba/numba/_build/results?buildId=16523&view=logs&j=bcb88f14-0280-5c18-73fb-fab3a418b655&t=83308b39-088b-5b5e-920a-7e5328b5d47f&l=10734 which requires fix from https://github.com/numba/numba/pull/9359

**Comment by sklam** [comment]
Rerunning failed tests after merge of #9359

**Review by stuartarchibald — APPROVED** [review]
Thanks again for the fix. I agree with https://github.com/numba/numba/pull/9360#issuecomment-1854761419, without #9359 the RVSDG build job should indeed fail, and with #9359 (now it has been merged) it should pass.

**Pull request merged**